boa/src/object/internal_methods/integer_indexed.rs

```diff
@@ -0,0 +1,385 @@
+        if let Some(value) = desc.value() {
+            integer_indexed_element_set(obj, index as usize, value, context)?
+        }
+
+        // vii. Return true.
+        Ok(true)
+    } else {
+        // 2. Return ! OrdinaryDefineOwnProperty(O, P, Desc).
+        super::ordinary_define_own_property(obj, key, desc, context)
+    }
+}
+
+/// Internal method `[[Get]]` for Integer-Indexed exotic objects.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-integer-indexed-exotic-objects-get-p-receiver
+#[inline]
+pub(crate) fn integer_indexed_exotic_get(
+    obj: &JsObject,
+    key: &PropertyKey,
+    receiver: JsValue,
+    context: &mut Context,
+) -> JsResult<JsValue> {
+    // 1. If Type(P) is String, then
+    // a. Let numericIndex be ! CanonicalNumericIndexString(P).
+    // b. If numericIndex is not undefined, then
+    if let PropertyKey::Index(index) = key {
+        // i. Return ! IntegerIndexedElementGet(O, numericIndex).
+        Ok(integer_indexed_element_get(obj, *index as usize).unwrap_or_default())
+    } else {
+        // 2. Return ? OrdinaryGet(O, P, Receiver).
+        super::ordinary_get(obj, key, receiver, context)
+    }
+}
+
+/// Internal method `[[Set]]` for Integer-Indexed exotic objects.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-integer-indexed-exotic-objects-set-p-v-receiver
+#[inline]
+pub(crate) fn integer_indexed_exotic_set(
+    obj: &JsObject,
+    key: PropertyKey,
+    value: JsValue,
+    receiver: JsValue,
+    context: &mut Context,
+) -> JsResult<bool> {
+    // 1. If Type(P) is String, then
+    // a. Let numericIndex be ! CanonicalNumericIndexString(P).
+    // b. If numericIndex is not undefined, then
+    if let PropertyKey::Index(index) = key {
+        // i. Perform ? IntegerIndexedElementSet(O, numericIndex, V).
+        integer_indexed_element_set(obj, index as usize, &value, context)?;
+
+        // ii. Return true.
+        Ok(true)
+    } else {
+        // 2. Return ? OrdinarySet(O, P, V, Receiver).
+        super::ordinary_set(obj, key, value, receiver, context)
+    }
+}
+
+/// Internal method `[[Delete]]` for Integer-Indexed exotic objects.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-integer-indexed-exotic-objects-delete-p
+#[inline]
+pub(crate) fn integer_indexed_exotic_delete(
+    obj: &JsObject,
+    key: &PropertyKey,
+    context: &mut Context,
+) -> JsResult<bool> {
+    // 1. If Type(P) is String, then
+    // a. Let numericIndex be ! CanonicalNumericIndexString(P).
+    // b. If numericIndex is not undefined, then
+    if let PropertyKey::Index(index) = key {
+        // i. If ! IsValidIntegerIndex(O, numericIndex) is false, return true; else return false.
+        Ok(!is_valid_integer_index(obj, *index as usize))
```

Comment:
This might be switched, no?
Because the comment can be negated to read:
`// i. If IsValidIntegerIndex(O, numericIndex) is true, return true; else return false.`
```suggestion
        // i. If ! IsValidIntegerIndex(O, numericIndex) is false, return true; else return false.
        Ok(is_valid_integer_index(obj, *index as usize))
```

Reply:
Ah, no, the `!` in the spec means the function call must not fail